src/test/java/org/jabref/logic/openoffice/oocsltext/CSLFormatUtilsTest.java

```diff
@@ -0,0 +1,724 @@
+package org.jabref.logic.openoffice.oocsltext;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.jabref.logic.citationstyle.CitationStyle;
+import org.jabref.logic.citationstyle.CitationStyleGenerator;
+import org.jabref.logic.citationstyle.CitationStyleOutputFormat;
+import org.jabref.logic.util.TestEntry;
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.database.BibDatabaseMode;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BibEntryTypesManager;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.model.openoffice.ootext.OOText;
+
+import de.undercouch.citeproc.output.Citation;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.jabref.logic.openoffice.oocsltext.CSLFormatUtils.generateAlphanumericCitation;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class CSLFormatUtilsTest {
+
+    private static final List<CitationStyle> STYLE_LIST = CitationStyle.discoverCitationStyles();
+
+    private final BibEntry testEntry = TestEntry.getTestEntry();
+    private final BibDatabaseContext context = new BibDatabaseContext(new BibDatabase(List.of(TestEntry.getTestEntry())));
+    private final BibEntryTypesManager bibEntryTypesManager = new BibEntryTypesManager();
+
+    /**
+     * Test to check transformation of raw, unsupported HTML into OO-ready HTML.
+     *
+     * @param input the raw HTML.
+     */
+    @ParameterizedTest
+    @MethodSource("rawHTMLProvider")
+    void ooHTMLTransformFromRawHTML(String input, String expected) {
+        String actual = CSLFormatUtils.transformHTML(input);
+        assertEquals(expected, actual);
+    }
+
+    static Stream<Arguments> rawHTMLProvider() {
+        return Stream.of(
+
+                // First three are general test cases for unescaping HTML entities
```

Comment:
Use `// region: ...` and then `// endregion`. Reason: Someone might add new test cases and might not update the commet.